nncf/torch/quantization/weights_compression.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict, Optional
+
+import torch
+from torch import nn
+
+from nncf.torch.layers import NNCF_WRAPPED_USER_MODULES_DICT
+from nncf.torch.layers import NNCFEmbedding
+from nncf.torch.layers import NNCFLinear
+from nncf.torch.quantization.quantize_functions import get_scale_zp_from_input_low_input_high
+
+
+class WeightsCompressor(nn.Module):
+    """Class provided fake quantize operation or dequantization of compressed weights in forward pass
+
+    Atributes:
+        zero_point: zero point in quantization scheme
+        scale: scale in quantizatin scheme
+    """
+
+    def __init__(self, zero_point, scale):
+        super().__init__()
+        self.zero_point = zero_point
+        self.scale = scale
+
+    def forward(self, input, op_arg):
+        if input.weight.dtype is torch.uint8:
+            w = input.weight.type(dtype=self.scale.dtype)
+            input.weight = (w - self.zero_point) * self.scale
+        elif not self.zero_point.dtype is torch.uint8:
+            axis = 0 if input.weight.shape[0] == self.scale.shape[0] else 1
+            input.weight = torch.fake_quantize_per_channel_affine(
+                input.weight, self.scale, self.zero_point, axis, 0, 255
+            )
+
+
+def _insert_pre_compression_operations(
+    module: nn.Module, allowed_types: Dict, compress_weights=False
+) -> Optional[nn.Module]:
+    """
+    Insets weights compression with dequantization or quantization pre operation for Linear and Embedding layers.
+
+    :param module: The module to insert the weights compression.
+    :param compress_weights: Enables real compression of weights in Linear and Embedding layers.
+        If False inserts pytorch torch.fake_quantize_per_channel_affine(),
+        else compress weights to int8 and inserts custom dequantization.
+    :param allowed_types: dist with pairs (allowed type for weights compression, dimention for scale and zero point)
```

Comment:
It's not just types. I think there can be a better name that would say something about compression dimension also. 
Maybe you even don't need this dict, just list of types, because `NNCFEmbedding` and `NNCFLinear` have this information in `target_weight_dim_for_compression`. Why not reuse this?

Reply:
Done